src/components/NucleotideEntropy.tsx

```diff
@@ -0,0 +1,706 @@
+import { colors } from '../widgets/common';
+import { mapLabelsToColors } from '../helpers/colors';
+import chroma from 'chroma-js';
+import Select, { CSSObjectWithLabel, StylesConfig } from 'react-select';
+import { getTicks } from '../helpers/ticks';
+import { formatDate } from '../widgets/VariantTimeDistributionLineChartInner';
+import { PercentageInput } from './PercentageInput';
+
+type Props = {
+  selector: LapisSelector;
+};
+
+type PositionProportion = {
+  position: string;
+  mutation: string | undefined;
+  original: string | undefined;
+  proportion: number;
+};
+
+type PositionEntropy = {
+  position: string;
+  proportions: PositionProportion[];
+  entropy: number;
+};
+
+type WeekEntropy = {
+  week: DateRange;
+  meanEntropy: number;
+};
+
+export type TransformedTime = {
+  [x: string]: string | number | undefined;
+  day: number | undefined;
+}[];
+
+type Data = {
+  positionEntropy: PositionEntropy[];
+  timeData: any;
+  sequenceType: SequenceType;
+  ticks: number[];
+};
+
+type Gene = {
+  name: string;
+  startPosition: number;
+  endPosition: number;
+  aaSeq: string;
+};
+
+export type GeneOption = {
+  value: string;
+  label: string;
+  startPosition: number;
+  endPosition: number;
+  aaSeq: string;
+  color: string;
+};
+
+const toolTipStyle = {
+  backgroundColor: 'white',
+  zIndex: 1,
+};
+
+const colorStyles: Partial<StylesConfig<any, true, any>> = {
+  control: (styles: CSSObjectWithLabel) => ({ ...styles, backgroundColor: 'white' }),
+  multiValue: (styles: CSSObjectWithLabel, { data }: { data: GeneOption }) => {
+    const color = chroma(data.color);
+    return {
+      ...styles,
+      backgroundColor: color.alpha(0.1).css(),
+    };
+  },
+  multiValueLabel: (styles: CSSObjectWithLabel, { data }: { data: GeneOption }) => ({
+    ...styles,
+    color: data.color,
+  }),
+  multiValueRemove: (styles: CSSObjectWithLabel, { data }: { data: GeneOption }) => {
+    return {
+          ...styles,
+          'color': data.color,
+          ':hover': {
+            backgroundColor: data.color,
+            color: 'white',
+            cursor: 'pointer',
+          },
+        };
+  },
+};
+
+function assignColorsToGeneOptions<T extends { label: string }>(options: T[]): (T & { color: string })[] {
+  const colors = mapLabelsToColors(options.map(o => o.label));
+  return options.map((o, i) => ({
+    ...o,
+    color: colors[i],
+  }));
+}
+
+const genes = jsonRefData.genes;
+!genes.map(o => o.name).includes('All') &&
+  genes.push({ name: 'All', startPosition: 0, endPosition: 29903, aaSeq: ''});
+
+const options: GeneOption[] = assignColorsToGeneOptions(genes.map(g => {
+  return {
+    value: g.name,
+    label: g.name,
+    startPosition: g.startPosition,
+    endPosition: g.endPosition,
+    aaSeq: g.aaSeq,
+    color: chroma.random().darken().hex()
+  };
+}));
+
+
+export const NucleotideEntropy = ({ selector }: Props) => {
+  const [plotType, setPlotType] = useState<string>('time');
```

Comment:
I think we can improve the typing (and naming) here:
```suggestion
  const [plotType, setPlotType] = useState<'time' | 'position'>('time');
```